src/main/java/htsjdk/samtools/BAMRecordCodec.java

```diff
@@ -113,11 +118,13 @@ public void encode(final SAMRecord alignment) {
         }
 
         int indexBin = 0;
-        if (alignment.getReferenceIndex() >= 0) {
+        if (alignment.getAlignmentStart() != SAMRecord.NO_ALIGNMENT_START) {
+            warnIfReferenceIsTooLargeForBinField(alignment);
+
             if (alignment.getIndexingBin() != null) {
                 indexBin = alignment.getIndexingBin();
             } else {
-                indexBin = alignment.computeIndexingBin();
+                indexBin = getUshort(alignment.computeIndexingBin());
             }
```

Comment:
I think it would be better to move this whole if/else stmt (lines 124-128), and it's companion the getUshort method, into SAMRecord as a method that is colocated with computeIndexingBin.

Reply:
Done.